### PR TITLE
notify users when they are given edit access to a work/collection

### LINKED ISSUE #1158: notify users when they are given edit access to a work
# Feature Request

The UI for granting users edit access to a work or collection (on the Settings page) should include a checkbox that, if checked, causes an email to be sent to any users who have gained edit access to the work/collection as a result of the form submission. 

The label for the checkbox should read: "send notification email"

The email should read:

> SUBJECT: Edit Access for {abbreviated work title}
> 
> Dear {name},
> 
> You have been given edit access to the following {work/collection} on ScholarSphere: 
>    - {abbreviated work title}: {link}
> 
> If you believe you have been given edit access in error, please contact {scholarsphere contact email}
>

### PR DESCRIPTION
**Pull request by banukutlu**
Closes #1158

### CODE REVIEW

File: spec/models/collection_spec.rb [hunk 1]
```diff
@@ -247,6 +248,7 @@
         uuid_ssi
         visibility_ssi
         work_type_ss
+        notify_editors_tesim
```
Comment (nicktiberi) [inline spec/models/collection_spec.rb:251]: Just curious why this key is here — is this something that is generated by Blacklight automatically somehow?

File: spec/forms/editors_form_spec.rb [hunk 2]
```diff
@@ -94,14 +95,26 @@
...
+
+              expect(mailer_spy).to have_received(:added_as_editor)
+              expect(mailer_spy).to have_received(:deliver_later)
+            end
+          end
 
-            expect(ActorMailer).to have_received(:with).with(actor: user.actor, resource: resource)
-            expect(ActorMailer).not_to have_received(:with).with(actor: existing_user.actor, resource: resource)
+          context 'when the send notification email is not checked' do
+            it 'sends an email to any new users' do
```
Comment (nicktiberi) [inline spec/forms/editors_form_spec.rb]:
```suggestion
            it 'does not send an email to any new users' do
```

**Review by nicktiberi — COMMENTED** [review]
LGTM! I just have a small suggestion and a question re: Solr fields.

File: spec/forms/editors_form_spec.rb (hunk 2 above)
Comment (banukutlu) [reply spec/forms/editors_form_spec.rb]: oops thanks for catching this :)

**Comment by ajkiessl** [comment]
@banukutlu This looks good to me!

**Review by ajkiessl — APPROVED** [review]

File: spec/models/collection_spec.rb (hunk 1 above)
Comment (ajkiessl) [reply spec/models/collection_spec.rb:251]: I think that gets set here: https://github.com/psu-libraries/scholarsphere/blob/main/app/schemas/default_schema.rb#L22

Comment (banukutlu) [reply spec/models/collection_spec.rb:251]: @nicktiberi, it looks like the work's [to_solr](https://github.com/psu-libraries/scholarsphere/blob/main/app/models/work.rb#L164-L166 )  and collection's [to_solr](https://github.com/psu-libraries/scholarsphere/blob/6198f2422d2d3130d894407e2f4163d9f147fa15/app/models/collection.rb#L133-L135) uses [SolrDocumentBuilder](https://github.com/psu-libraries/scholarsphere/blob/6198f2422d2d3130d894407e2f4163d9f147fa15/app/services/solr_document_builder.rb#L3)

gets each attribute of a resource to be indexed

https://github.com/psu-libraries/scholarsphere/blob/6198f2422d2d3130d894407e2f4163d9f147fa15/app/schemas/default_schema.rb#L12-L24

**Review by nicktiberi — APPROVED** [review]

**Pull request merged**